src/data/mystery-encounters/encounters/fiery-fallout-encounter.ts

```diff
@@ -0,0 +1,250 @@
+import { EncounterOptionMode, MysteryEncounterOptionBuilder } from "#app/data/mystery-encounters/mystery-encounter-option";
+import { applyDamageToPokemon, EnemyPartyConfig, generateModifierTypeOption, initBattleWithEnemyConfig, initCustomMovesForEncounter, leaveEncounterWithoutBattle, setEncounterExp, setEncounterRewards, transitionMysteryEncounterIntroVisuals } from "#app/data/mystery-encounters/utils/encounter-phase-utils";
+import { modifierTypes, } from "#app/modifier/modifier-type";
+import { MysteryEncounterType } from "#enums/mystery-encounter-type";
+import BattleScene from "../../../battle-scene";
+import IMysteryEncounter, { MysteryEncounterBuilder, MysteryEncounterTier, } from "../mystery-encounter";
+import { TypeRequirement } from "../mystery-encounter-requirements";
+import { Species } from "#enums/species";
+import { getPokemonSpecies } from "#app/data/pokemon-species";
+import { Gender } from "#app/data/gender";
+import { Type } from "#app/data/type";
+import { BattlerIndex } from "#app/battle";
+import { PokemonMove } from "#app/field/pokemon";
+import { Moves } from "#enums/moves";
+import { EncounterAnim, EncounterBattleAnim } from "#app/data/battle-anims";
+import { WeatherType } from "#app/data/weather";
+import { isNullOrUndefined, randSeedInt } from "#app/utils";
+import { StatusEffect } from "#app/data/status-effect";
+import { queueEncounterMessage } from "#app/data/mystery-encounters/utils/encounter-dialogue-utils";
+
+/** the i18n namespace for the encounter */
+const namespace = "mysteryEncounter:fieryFallout";
+
+/**
+ * Damage percentage taken when suffering the heat.
+ * Can be a number between `0` - `100`.
+ * The higher the more damage taken (100% = instant KO).
+ */
+const DAMAGE_PERCENTAGE: number = 20;
+
+/**
+ * Fiery Fallout encounter.
+ * @see {@link https://github.com/AsdarDevelops/PokeRogue-Events/issues/88 | GitHub Issue #88}
+ * @see For biome requirements check {@linkcode mysteryEncountersByBiome}
+ */
+export const FieryFalloutEncounter: IMysteryEncounter =
+  MysteryEncounterBuilder.withEncounterType(MysteryEncounterType.FIERY_FALLOUT)
+    .withEncounterTier(MysteryEncounterTier.COMMON)
+    .withSceneWaveRangeRequirement(40, 180) // waves 10 to 180
```

Comment:
```suggestion
    .withSceneWaveRangeRequirement(40, 180)
```

that comment feels a bit redundant